src/lsptoolshost/services/projectContextService.ts

```diff
@@ -58,6 +72,20 @@ export class ProjectContextService {
         this._contextChangeEmitter.fire({ uri, context });
     }
 
+    private async getVirtualCSharpUri(uri: vscode.Uri): Promise<vscode.Uri | undefined> {
+        const response = await vscode.commands.executeCommand<ProvideDynamicFileResponse>(
+            DynamicFileInfoHandler.provideDynamicFileInfoCommand,
+            new ProvideDynamicFileParams([uri.fsPath])
+        );
+
+        const responseUri = response.generatedFiles[0];
+        if (!responseUri) {
```

Comment:
Does this need an explicit check for `null` too?
https://github.com/dotnet/vscode-csharp/blob/d7439d3fa6a13800dc12f3ce719c1010c2cb38d3/src/razor/src/dynamicFile/dynamicFileInfoHandler.ts#L47

Reply:
The false-y check should cover null, undefined, and i think '' as well.